tests/test_endecoder.py

```diff
@@ -179,6 +185,14 @@ def test_endecodes_raises_exception(self):
         with self.assertRaises(decoder.BibDecodingError):
             decoder.decode_bibdata("@misc{I am not a correct bibtex{{}")
 
+    def test_endecode_preserves_type(self):
+        decoder = endecoder.EnDecoder()
+        entry = decoder.decode_bibdata(bibtex_raw0)
+
+        bibraw1 = decoder.encode_bibdata(
+            entry, ignore_fields=['title', 'note', 'abstract', 'journal'])
+
```

Comment:
wouldn't you want to re-decode bibraw1 at that point to make sure `type` is still present and "technical report"?

Reply:
Indeed, I think I left this one unfinished.